### PR TITLE
Section Migration: part 4 [do not merge]

### PR DESCRIPTION
**Pull request by domdinicola**

### CODE REVIEW

File: src/etools/applications/partners/migrations/0015_ref_number_year.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+def populate_agreement_ref_num(apps, schema):
+    import re
```
Comment (reinbach) [inline src/etools/applications/partners/migrations/0015_ref_number_year.py]: How come these imports are inside the function?

File: src/etools/applications/partners/migrations/0015_ref_number_year.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def populate_intervention_ref_num(apps, schema):
+    import re
+    from django.db import IntegrityError
+    Intervention = apps.get_model("partners", "Intervention")
+
+    for i in Intervention.objects.exclude(document_type="SSFA"):
+        find = re.findall(r'\d{4}', i.number)
+        try:
```
Comment (reinbach) [inline src/etools/applications/partners/migrations/0015_ref_number_year.py]: This section appears repetitive, you could move this into a function that takes `status_list` parameter. Granted it is just a data migration.

File: src/etools/applications/partners/migrations/0016_auto_20180717_0700.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('partners', '0015_ref_number_year'),
+    ]
+
+    operations = [
+        migrations.AlterField(
```
Comment (reinbach) [inline src/etools/applications/partners/migrations/0016_auto_20180717_0700.py]: You could merge this migration with `src/etools/applications/partners/migrations/0014_auto_20180716_1547.py`

File: src/etools/applications/partners/models.py [hunk 4]
```diff
@@ -2060,8 +2070,11 @@ def save(self, force_insert=False, **kwargs):
             # to create a reference number we need a pk
             super(Intervention, self).save()
             self.update_reference_number()
+        elif self.status == self.DRAFT:
+            self.update_reference_number()
 
-        self.update_ssfa_properties()
+        if save_from_agreement is False:
```
Comment (reinbach) [inline src/etools/applications/partners/models.py]: maybe `if not save_form_agreement:`  is more pythonic?

File: src/etools/applications/partners/migrations/0015_ref_number_year.py (hunk 1 above)
Comment (domdinicola) [reply src/etools/applications/partners/migrations/0015_ref_number_year.py]: I think that after the migration run in production we can remove the code in the datamigrations to speed up tests

Comment (domdinicola) [reply src/etools/applications/partners/migrations/0015_ref_number_year.py]: @robertavram any reason?

File: src/etools/applications/partners/migrations/0015_ref_number_year.py (hunk 2 above)
Comment (domdinicola) [reply src/etools/applications/partners/migrations/0015_ref_number_year.py]: I think that after the migration run in production we can remove the code in the datamigrations to speed up tests

**Pull request merged**